src/analytic/SRT2DSPECT/SRT2DSPECTReconstruction.cxx

```diff
@@ -0,0 +1,968 @@
+  num_segments_to_combine = num_segments_to_combine_v;
+  filter_wiener = filter_wiener_v;
+  filter_median = filter_median_v;
+  filter_gamma = filter_gamma_v;
+}
+
+Succeeded
+SRT2DSPECTReconstruction::actual_reconstruct(shared_ptr<DiscretisedDensity<3, float>> const& density_ptr)
+{
+
+  // perform SSRB
+    if (num_segments_to_combine>1)
+      {
+        const ProjDataInfoCylindrical& proj_data_info_cyl =
+          dynamic_cast<const ProjDataInfoCylindrical&>
+          (*proj_data_ptr->get_proj_data_info_sptr());
+
+        //  full_log << "SSRB combining " << num_segments_to_combine
+        //           << " segments in input file to a new segment 0\n" << std::endl;
+
+        shared_ptr<ProjDataInfo>
+          ssrb_info_sptr(SSRB(proj_data_info_cyl,
+                              num_segments_to_combine,
+                              1, 0,
+                              (num_segments_to_combine-1)/2 ));
+        shared_ptr<ProjData>
+          proj_data_to_SRT_ptr(new ProjDataInMemory (proj_data_ptr->get_exam_info_sptr(), ssrb_info_sptr));
+        SSRB(*proj_data_to_SRT_ptr, *proj_data_ptr);
+        proj_data_ptr = proj_data_to_SRT_ptr;
+      }
+    else
+      {
+        // just use the proj_data_ptr we have already
+      }  
+
+  // check if segment 0 has direct sinograms
+  {
+    const float tan_theta = proj_data_ptr->get_proj_data_info_sptr()->get_tantheta(Bin(0, 0, 0, 0));
+    if (fabs(tan_theta) > 1.E-4)
+      {
+        warning("SRT2DSPECT: segment 0 has non-zero tan(theta) %g", tan_theta);
+        return Succeeded::no;
+      }
+  }
+
+  float tangential_sampling;
+  // TODO make next type shared_ptr<ProjDataInfoCylindricalArcCorr> once we moved to boost::shared_ptr
+  // will enable us to get rid of a few of the ugly lines related to tangential_sampling below
+  shared_ptr<const ProjDataInfo> arc_corrected_proj_data_info_sptr;
+
+  // arc-correction if necessary
+  ArcCorrection arc_correction;
+  bool do_arc_correction = false;
+  if (!is_null_ptr(dynamic_pointer_cast<const ProjDataInfoCylindricalArcCorr>(proj_data_ptr->get_proj_data_info_sptr())))
+    {
+      // it's already arc-corrected
+      arc_corrected_proj_data_info_sptr = proj_data_ptr->get_proj_data_info_sptr()->create_shared_clone();
+      tangential_sampling = dynamic_cast<const ProjDataInfoCylindricalArcCorr&>(*proj_data_ptr->get_proj_data_info_sptr())
+                                .get_tangential_sampling();
+    }
+  else
+    {
+      // TODO arc-correct to voxel_size
+      if (arc_correction.set_up(proj_data_ptr->get_proj_data_info_sptr()->create_shared_clone()) == Succeeded::no)
+        return Succeeded::no;
+      do_arc_correction = true;
+      // TODO full_log
+      warning("SRT2DSPECT will arc-correct data first");
+      arc_corrected_proj_data_info_sptr = arc_correction.get_arc_corrected_proj_data_info_sptr();
+      tangential_sampling = arc_correction.get_arc_corrected_proj_data_info().get_tangential_sampling();
+    }
```

Comment:
I marked it as resolved because I commented the prior version and added the new one. However, I have not erased the comments yet, as I explained in previous message. This might hold for other issues which I state as resolved: they are ready in terms of code to be checked, but not in terms of format, i.e. comments/headers etc.

Reply:
I decided to unresolve conversations which are not ready in terms of comments removal, even if the code issue itself is resolved.
Hence, this conversation in terms of code it is resolved, but I hold it unresolved because of comments not having been removed.

Reply:
To correct the Codacy error: Variable 'tangential_sampling' is assigned a value that is never used.
I changed the new code to: 

auto pdi_sptr = dynamic_pointer_cast<const ProjDataInfoCylindricalArcCorr>(proj_data_ptr->get_proj_data_info_sptr());
if (!pdi_sptr)
{
    error("SPECT data should correspond to ProjDataInfoCylindricalArcCorr");
}